kernel/enter.c

```diff
@@ -446,15 +470,26 @@ int myst_enter_kernel(myst_kernel_args_t* args)
     }
     else
     {
-#ifdef MYST_ENABLE_EXT2FS
+        bool loaded_fs = false;
 
+#if defined(MYST_ENABLE_EXT2FS)
         /* setup and mount the EXT2 file system */
-        if (_setup_ext2(args->rootfs) != 0)
-            ERAISE(-EINVAL);
+        if (_setup_ext2(args->rootfs) == 0)
+        {
+            loaded_fs = true;
+        }
+#endif
 
-#else  /* MYST_ENABLE_EXT2FS */
-        myst_panic("ext2 not supported");
-#endif /* MYST_ENABLE_EXT2FS */
+#if defined(MYST_ENABLE_HOSTFS) && defined(MYST_DEBUG)
+        /* setup and mount the HOSTFS file system */
+        if (!loaded_fs && _setup_hostfs(args->rootfs) == 0)
+        {
+            loaded_fs = true;
+        }
+#endif
+
+        if (!loaded_fs)
+            myst_panic("failed to mount rootfs: %s", args->rootfs);
```

Comment:
Maybe more informative message here? Like: `Failed to mount the rootfs %s as either RAMFS (why?), EXT2 (why?), or HOSTFS (why?)`

Reply:
Good point. I cleaned all this up with a new function called **_get_fstype()** that determines the file system type and then we use a switch statement that is much cleaner. Also, I added error string output arguments to the setup functions so we can provide better error messages.